portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,94 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-
+import java.util.ArrayList;
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  
 
   @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String text = request.getParameter("text-input");
+
+    Entity taskEntity = new Entity("Comments");
+    
+    if (text != ""){
+        //do not add a comment if it is empty
+        taskEntity.setProperty("Comment", text);
+    }
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(taskEntity);
+
+    // commentsarray.add(text);
+    response.sendRedirect("/"); //redirects to main page
+  }
+
+
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello Juan!");
+    ArrayList<String> commentsarray = new ArrayList<String>();
+    Query query = new Query("Comments");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity : results.asIterable()) {
+        String comment = (String) entity.getProperty("Comment");
```

Comment:
This technically works because the only comments property you care about is the text/just a string, but theoretically you should create a Comment.java just like Task.java here: https://github.com/John256145/software-product-sprint/blob/master/walkthroughs/week-2-server/examples/todo-list/src/main/java/com/google/sps/data/Task.java This defines an interface between the server and client on how to parse/handle a Comments object. This becomes more crucial once you start supporting other comment features, like timestamps, authors, etc. so it's not just about handling a string anymore but more structured information. In doGet(), when you retrieve info from the Datastore, you then construct a Comment object with your data like this: https://github.com/John256145/software-product-sprint/blob/master/walkthroughs/week-2-server/examples/todo-list/src/main/java/com/google/sps/servlets/ListTasksServlet.java#L50 so that your client knows how to parse it (i.e. which property to use to retrieve the author's name for example) like this https://github.com/John256145/software-product-sprint/blob/master/walkthroughs/week-2-server/examples/todo-list/src/main/webapp/script.js#L31